<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Miscellaneous Topics | Point count data analysis: How to violate assumptions and get away with it</title>
  <meta name="description" content="This book provides material for the workshop ‘Analysis of point-count data in the presence of variable survey methodologies and detection error’ at the AOS 2019 conference." />
  <meta name="generator" content="bookdown 0.11 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Miscellaneous Topics | Point count data analysis: How to violate assumptions and get away with it" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="http://peter.solymos.org/qpad-book/" />
  <meta property="og:image" content="http://peter.solymos.org/qpad-book/images/cover.png" />
  <meta property="og:description" content="This book provides material for the workshop ‘Analysis of point-count data in the presence of variable survey methodologies and detection error’ at the AOS 2019 conference." />
  <meta name="github-repo" content="psolymos/qpad-book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Miscellaneous Topics | Point count data analysis: How to violate assumptions and get away with it" />
  
  <meta name="twitter:description" content="This book provides material for the workshop ‘Analysis of point-count data in the presence of variable survey methodologies and detection error’ at the AOS 2019 conference." />
  <meta name="twitter:image" content="http://peter.solymos.org/qpad-book/images/cover.png" />

<meta name="author" content="Peter Solymos" />


<meta name="date" content="2019-06-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="roadsides.html">

<script src="libs/jquery/jquery.min.js"></script>
<link href="libs/gitbook/css/style.css" rel="stylesheet" />
<link href="libs/gitbook/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">QPAD Book</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#about-the-book-and-the-course"><i class="fa fa-check"></i>About the book and the course</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#about-the-author"><i class="fa fa-check"></i>About the author</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#installing-r-and-rstudio"><i class="fa fa-check"></i>Installing R and RStudio</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#installing-required-packages"><i class="fa fa-check"></i>Installing required packages</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#installing-the-book"><i class="fa fa-check"></i>Installing the book</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-this-works"><i class="fa fa-check"></i>How this works</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#acknowledgments"><i class="fa fa-check"></i>Acknowledgments</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#design-based-approaches"><i class="fa fa-check"></i><b>1.1</b> Design-based approaches</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#model-based-approaches"><i class="fa fa-check"></i><b>1.2</b> Model-based approaches</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#our-approach"><i class="fa fa-check"></i><b>1.3</b> Our approach</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="pcdata.html"><a href="pcdata.html"><i class="fa fa-check"></i><b>2</b> Organizing and Processing Point Count Data</a><ul>
<li class="chapter" data-level="2.1" data-path="pcdata.html"><a href="pcdata.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="pcdata.html"><a href="pcdata.html#prerequisites"><i class="fa fa-check"></i><b>2.2</b> Prerequisites</a></li>
<li class="chapter" data-level="2.3" data-path="pcdata.html"><a href="pcdata.html#rbasics"><i class="fa fa-check"></i><b>2.3</b> R basics</a></li>
<li class="chapter" data-level="2.4" data-path="pcdata.html"><a href="pcdata.html#josm-data-set"><i class="fa fa-check"></i><b>2.4</b> JOSM data set</a></li>
<li class="chapter" data-level="2.5" data-path="pcdata.html"><a href="pcdata.html#cross-tabulating-species-counts"><i class="fa fa-check"></i><b>2.5</b> Cross tabulating species counts</a></li>
<li class="chapter" data-level="2.6" data-path="pcdata.html"><a href="pcdata.html#joining-species-data-with-predictors"><i class="fa fa-check"></i><b>2.6</b> Joining species data with predictors</a></li>
<li class="chapter" data-level="2.7" data-path="pcdata.html"><a href="pcdata.html#explore-predictor-variables"><i class="fa fa-check"></i><b>2.7</b> Explore predictor variables</a></li>
<li class="chapter" data-level="2.8" data-path="pcdata.html"><a href="pcdata.html#derived-variables"><i class="fa fa-check"></i><b>2.8</b> Derived variables</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="regression.html"><a href="regression.html"><i class="fa fa-check"></i><b>3</b> A Primer in Regression Techniques</a><ul>
<li class="chapter" data-level="3.1" data-path="regression.html"><a href="regression.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="regression.html"><a href="regression.html#prerequisites-1"><i class="fa fa-check"></i><b>3.2</b> Prerequisites</a></li>
<li class="chapter" data-level="3.3" data-path="regression.html"><a href="regression.html#poisson-null-model"><i class="fa fa-check"></i><b>3.3</b> Poisson null model</a></li>
<li class="chapter" data-level="3.4" data-path="regression.html"><a href="regression.html#exploring-covariates"><i class="fa fa-check"></i><b>3.4</b> Exploring covariates</a></li>
<li class="chapter" data-level="3.5" data-path="regression.html"><a href="regression.html#single-covariate"><i class="fa fa-check"></i><b>3.5</b> Single covariate</a></li>
<li class="chapter" data-level="3.6" data-path="regression.html"><a href="regression.html#additive-model"><i class="fa fa-check"></i><b>3.6</b> Additive model</a></li>
<li class="chapter" data-level="3.7" data-path="regression.html"><a href="regression.html#nonlinear-terms"><i class="fa fa-check"></i><b>3.7</b> Nonlinear terms</a></li>
<li class="chapter" data-level="3.8" data-path="regression.html"><a href="regression.html#categorical-variables"><i class="fa fa-check"></i><b>3.8</b> Categorical variables</a></li>
<li class="chapter" data-level="3.9" data-path="regression.html"><a href="regression.html#multiple-main-effects"><i class="fa fa-check"></i><b>3.9</b> Multiple main effects</a></li>
<li class="chapter" data-level="3.10" data-path="regression.html"><a href="regression.html#interaction"><i class="fa fa-check"></i><b>3.10</b> Interaction</a></li>
<li class="chapter" data-level="3.11" data-path="regression.html"><a href="regression.html#different-error-distributions"><i class="fa fa-check"></i><b>3.11</b> Different error distributions</a></li>
<li class="chapter" data-level="3.12" data-path="regression.html"><a href="regression.html#count-duration-effects"><i class="fa fa-check"></i><b>3.12</b> Count duration effects</a></li>
<li class="chapter" data-level="3.13" data-path="regression.html"><a href="regression.html#count-radius-effects"><i class="fa fa-check"></i><b>3.13</b> Count radius effects</a></li>
<li class="chapter" data-level="3.14" data-path="regression.html"><a href="regression.html#offsets"><i class="fa fa-check"></i><b>3.14</b> Offsets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="behavior.html"><a href="behavior.html"><i class="fa fa-check"></i><b>4</b> Behavioral Complexities</a><ul>
<li class="chapter" data-level="4.1" data-path="behavior.html"><a href="behavior.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="behavior.html"><a href="behavior.html#prerequisites-2"><i class="fa fa-check"></i><b>4.2</b> Prerequisites</a></li>
<li class="chapter" data-level="4.3" data-path="behavior.html"><a href="behavior.html#birds-in-the-forest"><i class="fa fa-check"></i><b>4.3</b> Birds in the forest</a></li>
<li class="chapter" data-level="4.4" data-path="behavior.html"><a href="behavior.html#survival-model"><i class="fa fa-check"></i><b>4.4</b> Survival model</a></li>
<li class="chapter" data-level="4.5" data-path="behavior.html"><a href="behavior.html#vocalization-events"><i class="fa fa-check"></i><b>4.5</b> Vocalization events</a></li>
<li class="chapter" data-level="4.6" data-path="behavior.html"><a href="behavior.html#removal-model"><i class="fa fa-check"></i><b>4.6</b> Removal model</a><ul>
<li class="chapter" data-level="4.6.1" data-path="behavior.html"><a href="behavior.html#real-data"><i class="fa fa-check"></i><b>4.6.1</b> Real data</a></li>
<li class="chapter" data-level="4.6.2" data-path="behavior.html"><a href="behavior.html#time-invariant-conventional-model"><i class="fa fa-check"></i><b>4.6.2</b> Time-invariant conventional model</a></li>
<li class="chapter" data-level="4.6.3" data-path="behavior.html"><a href="behavior.html#time-varying-conventional-removal-model"><i class="fa fa-check"></i><b>4.6.3</b> Time-varying conventional removal model</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="behavior.html"><a href="behavior.html#finite-mixtures"><i class="fa fa-check"></i><b>4.7</b> Finite mixtures</a><ul>
<li class="chapter" data-level="4.7.1" data-path="behavior.html"><a href="behavior.html#time-invariant-finite-mixture-removal-model"><i class="fa fa-check"></i><b>4.7.1</b> Time-invariant finite mixture removal model</a></li>
<li class="chapter" data-level="4.7.2" data-path="behavior.html"><a href="behavior.html#time-varying-finite-mixture-removal-models"><i class="fa fa-check"></i><b>4.7.2</b> Time-varying finite mixture removal models</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="behavior.html"><a href="behavior.html#let-the-best-model-win"><i class="fa fa-check"></i><b>4.8</b> Let the best model win</a></li>
<li class="chapter" data-level="4.9" data-path="behavior.html"><a href="behavior.html#estimating-abundance"><i class="fa fa-check"></i><b>4.9</b> Estimating abundance</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="detection.html"><a href="detection.html"><i class="fa fa-check"></i><b>5</b> The Detection Process</a></li>
<li class="chapter" data-level="6" data-path="recordings.html"><a href="recordings.html"><i class="fa fa-check"></i><b>6</b> Dealing with Recordings</a></li>
<li class="chapter" data-level="7" data-path="assumptions.html"><a href="assumptions.html"><i class="fa fa-check"></i><b>7</b> A Closer Look at Assumptions</a></li>
<li class="chapter" data-level="8" data-path="roadsides.html"><a href="roadsides.html"><i class="fa fa-check"></i><b>8</b> Understanding Roadside Surveys</a></li>
<li class="chapter" data-level="9" data-path="extras.html"><a href="extras.html"><i class="fa fa-check"></i><b>9</b> Miscellaneous Topics</a><ul>
<li class="chapter" data-level="" data-path="extras.html"><a href="extras.html#these-are-just-reminders-to-be-deleted-later"><i class="fa fa-check"></i>These are just reminders, to be deleted later</a></li>
<li class="chapter" data-level="9.1" data-path="extras.html"><a href="extras.html#binomial-model-and-censoring"><i class="fa fa-check"></i><b>9.1</b> Binomial model and censoring</a></li>
<li class="chapter" data-level="9.2" data-path="extras.html"><a href="extras.html#optimal-partitioning"><i class="fa fa-check"></i><b>9.2</b> Optimal partitioning</a></li>
<li class="chapter" data-level="9.3" data-path="extras.html"><a href="extras.html#optilevels"><i class="fa fa-check"></i><b>9.3</b> Optilevels</a></li>
<li class="chapter" data-level="9.4" data-path="extras.html"><a href="extras.html#n-mixture-models"><i class="fa fa-check"></i><b>9.4</b> N-mixture models</a></li>
<li class="chapter" data-level="9.5" data-path="extras.html"><a href="extras.html#estimating-abundance-1"><i class="fa fa-check"></i><b>9.5</b> Estimating abundance</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="http://peter.solymos.org/" target="blank">P&eacute;ter S&oacute;lymos</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Point count data analysis: How to violate assumptions and get away with it</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="extras" class="section level1">
<h1><span class="header-section-number">Chapter 9</span> Miscellaneous Topics</h1>
<p>model selection and conditional likelihood</p>
<p>variance/bias trade off</p>
<p>error propagation</p>
<p>MCMC?</p>
<p>N-mixture ideas</p>
<p>phylogenetic and life history/trait stuff</p>
<p>PIF methods</p>
<div id="these-are-just-reminders-to-be-deleted-later" class="section level2 unnumbered">
<h2>These are just reminders, to be deleted later</h2>
<p>You can label chapter and section titles using <code>{#label}</code> after them, e.g., we can reference Chapter <a href="intro.html#intro">1</a>. If you do not manually label them, there will be automatic labels anyway.</p>
<p>Figures and tables with captions will be placed in <code>figure</code> and <code>table</code> environments, respectively.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">4</span>, <span class="fl">.1</span>, <span class="fl">.1</span>))
<span class="kw">plot</span>(pressure, <span class="dt">type =</span> <span class="st">&#39;b&#39;</span>, <span class="dt">pch =</span> <span class="dv">19</span>)</code></pre>
<div class="figure" style="text-align: center"><span id="fig:nice-fig"></span>
<img src="qpad-book_files/figure-html/nice-fig-1.png" alt="Here is a nice figure!" width="80%" />
<p class="caption">
Figure 9.1: Here is a nice figure!
</p>
</div>
<p>Reference a figure by its code chunk label with the <code>fig:</code> prefix, e.g., see Figure <a href="extras.html#fig:nice-fig">9.1</a>. Similarly, you can reference tables generated from <code>knitr::kable()</code>, e.g., see Table <a href="extras.html#tab:nice-tab">9.1</a>.</p>
<pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw">kable</span>(
  <span class="kw">head</span>(iris, <span class="dv">20</span>), <span class="dt">caption =</span> <span class="st">&#39;Here is a nice table!&#39;</span>,
  <span class="dt">booktabs =</span> <span class="ot">TRUE</span>
)</code></pre>
<table>
<caption><span id="tab:nice-tab">Table 9.1: </span>Here is a nice table!</caption>
<thead>
<tr class="header">
<th align="right">Sepal.Length</th>
<th align="right">Sepal.Width</th>
<th align="right">Petal.Length</th>
<th align="right">Petal.Width</th>
<th align="left">Species</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">5.1</td>
<td align="right">3.5</td>
<td align="right">1.4</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">4.9</td>
<td align="right">3.0</td>
<td align="right">1.4</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">4.7</td>
<td align="right">3.2</td>
<td align="right">1.3</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">4.6</td>
<td align="right">3.1</td>
<td align="right">1.5</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">5.0</td>
<td align="right">3.6</td>
<td align="right">1.4</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">5.4</td>
<td align="right">3.9</td>
<td align="right">1.7</td>
<td align="right">0.4</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">4.6</td>
<td align="right">3.4</td>
<td align="right">1.4</td>
<td align="right">0.3</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">5.0</td>
<td align="right">3.4</td>
<td align="right">1.5</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">4.4</td>
<td align="right">2.9</td>
<td align="right">1.4</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">4.9</td>
<td align="right">3.1</td>
<td align="right">1.5</td>
<td align="right">0.1</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">5.4</td>
<td align="right">3.7</td>
<td align="right">1.5</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">4.8</td>
<td align="right">3.4</td>
<td align="right">1.6</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">4.8</td>
<td align="right">3.0</td>
<td align="right">1.4</td>
<td align="right">0.1</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">4.3</td>
<td align="right">3.0</td>
<td align="right">1.1</td>
<td align="right">0.1</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">5.8</td>
<td align="right">4.0</td>
<td align="right">1.2</td>
<td align="right">0.2</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">5.7</td>
<td align="right">4.4</td>
<td align="right">1.5</td>
<td align="right">0.4</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">5.4</td>
<td align="right">3.9</td>
<td align="right">1.3</td>
<td align="right">0.4</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">5.1</td>
<td align="right">3.5</td>
<td align="right">1.4</td>
<td align="right">0.3</td>
<td align="left">setosa</td>
</tr>
<tr class="odd">
<td align="right">5.7</td>
<td align="right">3.8</td>
<td align="right">1.7</td>
<td align="right">0.3</td>
<td align="left">setosa</td>
</tr>
<tr class="even">
<td align="right">5.1</td>
<td align="right">3.8</td>
<td align="right">1.5</td>
<td align="right">0.3</td>
<td align="left">setosa</td>
</tr>
</tbody>
</table>
</div>
<div id="binomial-model-and-censoring" class="section level2">
<h2><span class="header-section-number">9.1</span> Binomial model and censoring</h2>
<p>Try cloglog with a rare species, like BOCH</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co">#spp &lt;- &quot;OVEN&quot; # which species</span>
spp &lt;-<span class="st"> &quot;BOCH&quot;</span> <span class="co"># which species</span>
<span class="co">#spp &lt;- &quot;CAWA&quot; # which species</span>

x &lt;-<span class="st"> </span><span class="kw">data.frame</span>(
  josm<span class="op">$</span>surveys, 
  <span class="dt">y=</span><span class="kw">as.numeric</span>(ytot[<span class="kw">rownames</span>(x), spp]))
x<span class="op">$</span>y01 &lt;-<span class="st"> </span><span class="kw">ifelse</span>(x<span class="op">$</span>y <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>)

<span class="kw">table</span>(x<span class="op">$</span>y)


mP &lt;-<span class="st"> </span><span class="kw">glm</span>(y <span class="op">~</span><span class="st"> </span>Decid <span class="op">*</span><span class="st"> </span>ConifWet, x, <span class="dt">family=</span>poisson)
mBc &lt;-<span class="st"> </span><span class="kw">glm</span>(y01 <span class="op">~</span><span class="st"> </span>Decid <span class="op">*</span><span class="st"> </span>ConifWet, x, <span class="dt">family=</span><span class="kw">binomial</span>(<span class="st">&quot;cloglog&quot;</span>))
mBl &lt;-<span class="st"> </span><span class="kw">glm</span>(y01 <span class="op">~</span><span class="st"> </span>Decid <span class="op">*</span><span class="st"> </span>ConifWet, x, <span class="dt">family=</span><span class="kw">binomial</span>(<span class="st">&quot;logit&quot;</span>))

<span class="kw">coef</span>(mP)
<span class="kw">coef</span>(mBc)
<span class="kw">coef</span>(mBl)

<span class="kw">plot</span>(<span class="kw">fitted</span>(mBc) <span class="op">~</span><span class="st"> </span><span class="kw">fitted</span>(mP), <span class="dt">col=</span><span class="dv">4</span>, 
  <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(<span class="kw">fitted</span>(mP))), <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(<span class="kw">fitted</span>(mP))))
<span class="kw">points</span>(<span class="kw">exp</span>(<span class="kw">model.matrix</span>(mBc) <span class="op">%*%</span><span class="st"> </span><span class="kw">coef</span>(mBc)) <span class="op">~</span><span class="st"> </span><span class="kw">fitted</span>(mP), <span class="dt">col=</span><span class="dv">2</span>)
<span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</code></pre>
</div>
<div id="optimal-partitioning" class="section level2">
<h2><span class="header-section-number">9.2</span> Optimal partitioning</h2>
<pre class="sourceCode r"><code class="sourceCode r">oc &lt;-<span class="st"> </span><span class="kw">opticut</span>(<span class="kw">as.matrix</span>(ytot) <span class="op">~</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">strata =</span> x<span class="op">$</span>HAB, <span class="dt">dist=</span><span class="st">&quot;poisson&quot;</span>)
<span class="kw">plot</span>(oc)</code></pre>
</div>
<div id="optilevels" class="section level2">
<h2><span class="header-section-number">9.3</span> Optilevels</h2>
<p>When we have categorical or compositional (when e.g. proportions add up to 1, also called
the unit sum constraint) data,
we often want to simplify and merge classes or add up columns.
We can do this based on the structural understanding of these land cover classes
(call all treed classes Forest, like what we did for <code>FOR</code>, <code>WET</code> and <code>AHF</code>).</p>
<p>Alternatively, we can let the data (the birds) tell us how to merge the classes.
The algorithm does the following:</p>
<ol style="list-style-type: decimal">
<li>fit model with all classes,</li>
<li>order estimates for each class from smallest to largest,</li>
<li>merge classes that are near each others, 2 at a time, moving from smallest to largest,</li>
<li>compare <span class="math inline">\(\Delta\)</span>AIC or <span class="math inline">\(\Delta\)</span>BIC values for the merged models and pick the smallest,</li>
<li>treat this best merged model as an input in step 1 and star over until <span class="math inline">\(\Delta\)</span> is negative (no improvement).</li>
</ol>
<p>Here is the code for simplifying categories using the <code>opticut::optilevels</code> function:</p>
<pre class="sourceCode r"><code class="sourceCode r">M &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span>HAB<span class="dv">-1</span>, x)
<span class="kw">colnames</span>(M) &lt;-<span class="st"> </span><span class="kw">levels</span>(x<span class="op">$</span>HAB)
ol1 &lt;-<span class="st"> </span><span class="kw">optilevels</span>(x<span class="op">$</span>y, M, <span class="dt">dist=</span><span class="st">&quot;poisson&quot;</span>)
<span class="kw">sort</span>(<span class="kw">exp</span>(<span class="kw">coef</span>(<span class="kw">bestmodel</span>(ol1))))
<span class="co">## estimates</span>
<span class="kw">exp</span>(ol1<span class="op">$</span>coef)
<span class="co">## optimal classification</span>
ol1<span class="op">$</span>rank
<span class="kw">data.frame</span>(<span class="dt">combined_levels=</span>ol1<span class="op">$</span>levels[[<span class="kw">length</span>(ol1<span class="op">$</span>levels)]])</code></pre>
<p>Here is the code for simplifying compositional data:</p>
<pre class="sourceCode r"><code class="sourceCode r">ol2 &lt;-<span class="st"> </span><span class="kw">optilevels</span>(x<span class="op">$</span>y, x[,cn], <span class="dt">dist=</span><span class="st">&quot;poisson&quot;</span>)
<span class="kw">sort</span>(<span class="kw">exp</span>(<span class="kw">coef</span>(<span class="kw">bestmodel</span>(ol2))))
<span class="co">## estimates</span>
<span class="kw">exp</span>(ol2<span class="op">$</span>coef)
<span class="co">## optimal classification</span>
ol2<span class="op">$</span>rank
<span class="kw">head</span>(<span class="kw">groupSums</span>(<span class="kw">as.matrix</span>(x[,cn]), <span class="dv">2</span>, ol2<span class="op">$</span>levels[[<span class="kw">length</span>(ol2<span class="op">$</span>levels)]]))</code></pre>
</div>
<div id="n-mixture-models" class="section level2">
<h2><span class="header-section-number">9.4</span> N-mixture models</h2>
</div>
<div id="estimating-abundance-1" class="section level2">
<h2><span class="header-section-number">9.5</span> Estimating abundance</h2>
<p>Exponential model, bSims data</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">1</span>)
phi &lt;-<span class="st"> </span><span class="fl">0.5</span>
Den &lt;-<span class="st"> </span><span class="dv">1</span>
l &lt;-<span class="st"> </span><span class="kw">bsims_init</span>()
a &lt;-<span class="st"> </span><span class="kw">bsims_populate</span>(l, <span class="dt">density=</span>Den)
b &lt;-<span class="st"> </span><span class="kw">bsims_animate</span>(a, <span class="dt">vocal_rate=</span>phi, <span class="dt">move_rate=</span><span class="dv">0</span>)

tint &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="dv">5</span>
(tr &lt;-<span class="st"> </span><span class="kw">bsims_transcribe</span>(b, <span class="dt">tint=</span>tint))</code></pre>
<pre><code>## bSims transcript
##   1 km x 1 km
##   stratification: H
##   total abundance: 104
##   total duration: 10
##   detected: 104 heard
##   1st inds. [0-1, 1-2, 2-3, 3-4, 4-5 min] [0+ m]</code></pre>
<p>Multiple-visit stuff for bSims: the counting of new individuals
resets for each interval (also: needs equal intervals)</p>
<pre class="sourceCode r"><code class="sourceCode r">tr<span class="op">$</span>visits</code></pre>
<pre><code>##     0-1min 1-2min 2-3min 3-4min 4-5min
## 0+m     35     48     44     43     45</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">v &lt;-<span class="st"> </span><span class="kw">get_events</span>(b, <span class="dt">vocal_only=</span><span class="ot">TRUE</span>)
v &lt;-<span class="st"> </span>v[v<span class="op">$</span>t <span class="op">&lt;=</span><span class="st"> </span><span class="kw">max</span>(tint),]
v1 &lt;-<span class="st"> </span>v[<span class="op">!</span><span class="kw">duplicated</span>(v<span class="op">$</span>i),]

tmp &lt;-<span class="st"> </span>v1
tmp<span class="op">$</span>o &lt;-<span class="st"> </span><span class="kw">seq_len</span>(<span class="kw">nrow</span>(v1))
<span class="kw">plot</span>(o <span class="op">~</span><span class="st"> </span>t, tmp, <span class="dt">type=</span><span class="st">&quot;n&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Individuals&quot;</span>,
  <span class="dt">main=</span><span class="st">&quot;Vocalization events&quot;</span>, 
  <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="kw">nrow</span>(b<span class="op">$</span>nests)), <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="kw">max</span>(tint)))
<span class="cf">for</span> (i <span class="cf">in</span> tmp<span class="op">$</span>o) {
  tmp2 &lt;-<span class="st"> </span>v[v<span class="op">$</span>i <span class="op">==</span><span class="st"> </span>v1<span class="op">$</span>i[i],]
  <span class="kw">lines</span>(<span class="kw">c</span>(tmp2<span class="op">$</span>t[<span class="dv">1</span>], <span class="kw">max</span>(tint)), <span class="kw">c</span>(i,i), <span class="dt">col=</span><span class="st">&quot;grey&quot;</span>)
  <span class="kw">points</span>(tmp2<span class="op">$</span>t, <span class="kw">rep</span>(i, <span class="kw">nrow</span>(tmp2)), <span class="dt">cex=</span><span class="fl">0.5</span>)
  <span class="kw">points</span>(tmp2<span class="op">$</span>t[<span class="dv">1</span>], i, <span class="dt">pch=</span><span class="dv">19</span>, <span class="dt">cex=</span><span class="fl">0.5</span>)
}</code></pre>
<p><img src="qpad-book_files/figure-html/unnamed-chunk-108-1.png" width="672" /></p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(o <span class="op">~</span><span class="st"> </span>t, tmp, <span class="dt">type=</span><span class="st">&quot;n&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Individuals&quot;</span>,
  <span class="dt">main=</span><span class="st">&quot;Vocalization events&quot;</span>, 
  <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="kw">nrow</span>(b<span class="op">$</span>nests)), <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="kw">max</span>(tint)))
<span class="cf">for</span> (j <span class="cf">in</span> <span class="kw">seq_along</span>(tint)) {
  ii &lt;-<span class="st"> </span><span class="cf">if</span> (j <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)
    <span class="kw">c</span>(<span class="dv">0</span>, tint[j]) <span class="cf">else</span> <span class="kw">c</span>(tint[j<span class="dv">-1</span>], tint[j])
  vv &lt;-<span class="st"> </span>v[v<span class="op">$</span>t <span class="op">&gt;</span><span class="st"> </span>ii[<span class="dv">1</span>] <span class="op">&amp;</span><span class="st"> </span>v<span class="op">$</span>t <span class="op">&lt;=</span><span class="st"> </span>ii[<span class="dv">2</span>],]
  tmp &lt;-<span class="st"> </span>vv[<span class="op">!</span><span class="kw">duplicated</span>(vv<span class="op">$</span>i),]
  tmp<span class="op">$</span>o &lt;-<span class="st"> </span><span class="kw">seq_len</span>(<span class="kw">nrow</span>(tmp))
  <span class="cf">if</span> (<span class="kw">nrow</span>(tmp)) {
    <span class="cf">for</span> (i <span class="cf">in</span> tmp<span class="op">$</span>o) {
      tmp2 &lt;-<span class="st"> </span>vv[vv<span class="op">$</span>i <span class="op">==</span><span class="st"> </span>tmp<span class="op">$</span>i[i],]
      <span class="kw">lines</span>(<span class="kw">c</span>(tmp2<span class="op">$</span>t[<span class="dv">1</span>], ii[<span class="dv">2</span>]), <span class="kw">c</span>(i,i), <span class="dt">col=</span><span class="st">&quot;grey&quot;</span>)
      <span class="kw">points</span>(tmp2<span class="op">$</span>t, <span class="kw">rep</span>(i, <span class="kw">nrow</span>(tmp2)), <span class="dt">cex=</span><span class="fl">0.5</span>)
      <span class="kw">points</span>(tmp2<span class="op">$</span>t[<span class="dv">1</span>], i, <span class="dt">pch=</span><span class="dv">19</span>, <span class="dt">cex=</span><span class="fl">0.5</span>)
    }
  }
}</code></pre>
<p><img src="qpad-book_files/figure-html/unnamed-chunk-108-2.png" width="672" /></p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(unmarked)</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## Loading required package: Rcpp</code></pre>
<pre><code>## Loading required package: reshape2</code></pre>
<pre><code>## 
## Attaching package: &#39;unmarked&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:lme4&#39;:
## 
##     ranef</code></pre>
<pre><code>## The following objects are masked from &#39;package:nlme&#39;:
## 
##     getData, ranef</code></pre>
<pre><code>## The following objects are masked from &#39;package:raster&#39;:
## 
##     getData, projection</code></pre>
<pre><code>## The following object is masked from &#39;package:sp&#39;:
## 
##     coordinates</code></pre>
<pre><code>## The following object is masked from &#39;package:stats4&#39;:
## 
##     mle</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">f &lt;-<span class="st"> </span><span class="cf">function</span>() {
  a &lt;-<span class="st"> </span><span class="kw">bsims_populate</span>(l, <span class="dt">density=</span>Den)
  b &lt;-<span class="st"> </span><span class="kw">bsims_animate</span>(a, <span class="dt">vocal_rate=</span>phi, <span class="dt">move_rate=</span><span class="dv">0</span>)
  tr &lt;-<span class="st"> </span><span class="kw">bsims_transcribe</span>(b, <span class="dt">tint=</span>tint)
  <span class="kw">drop</span>(tr<span class="op">$</span>visits)
}

Den &lt;-<span class="st"> </span><span class="fl">0.01</span>

<span class="co">#ymx &lt;- tr$visits</span>
(ymx &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">replicate</span>(<span class="dv">10</span>, <span class="kw">f</span>())))</code></pre>
<pre><code>##       0-1min 1-2min 2-3min 3-4min 4-5min
##  [1,]      0      0      0      0      0
##  [2,]      0      1      0      0      1
##  [3,]      0      2      2      1      0
##  [4,]      1      1      0      1      0
##  [5,]      0      0      0      0      0
##  [6,]      0      0      0      0      0
##  [7,]      2      2      0      1      0
##  [8,]      1      1      0      0      0
##  [9,]      0      0      0      0      0
## [10,]      0      1      0      1      0</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co">## highly dependent on K when Den is higher</span>
nmix &lt;-<span class="st"> </span><span class="kw">pcount</span>(<span class="op">~</span><span class="dv">1</span> <span class="op">~</span><span class="dv">1</span>, <span class="kw">unmarkedFramePCount</span>(<span class="dt">y=</span>ymx), <span class="dt">K=</span><span class="dv">1000</span>)
<span class="kw">coef</span>(nmix)</code></pre>
<pre><code>## lam(Int)   p(Int) 
## -0.02602 -0.44730</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plogis</span>(<span class="kw">coef</span>(nmix)[<span class="dv">2</span>])</code></pre>
<pre><code>## p(Int) 
##   0.39</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">exp</span>(<span class="kw">coef</span>(nmix)[<span class="dv">1</span>])</code></pre>
<pre><code>## lam(Int) 
##   0.9743</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">Den <span class="op">*</span><span class="st"> </span><span class="dv">100</span></code></pre>
<pre><code>## [1] 1</code></pre>

<div id="refs" class="references">
<div>
<p>Barker, Nicole K. S., Patricia C. Fontaine, Steve G. Cumming, Diana Stralberg, Alana Westwood, Erin M. Bayne, Péter Sólymos, Fiona K. A. Schmiegelow, Samantha J. Song, and D. J. Rugg. 2015. “Ecological Monitoring Through Harmonizing Existing Data: Lessons from the Boreal Avian Modelling Project.” <em>Wildlife Society Bulletin</em> 39: 480–87. <a href="https://doi.org/10.1650/CONDOR-14-108.1">https://doi.org/10.1650/CONDOR-14-108.1</a>.</p>
</div>
<div>
<p>Mahon, C. Lisa, Gillian Holloway, Péter Sólymos, Steve G. Cumming, Erin M. Bayne, Fiona K. A. Schmiegelow, and Samantha J. Song. 2016. “Community Structure and Niche Characteristics of Upland and Lowland Western Boreal Birds at Multiple Spatial Scales.” <em>Forest Ecology and Management</em> 361: 99–116. <a href="https://doi.org/10.1016/j.foreco.2015.11.007">https://doi.org/10.1016/j.foreco.2015.11.007</a>.</p>
</div>
<div>
<p>Matsuoka, S. M., C. L. Mahon, C. M. Handel, Péter Sólymos, E. M. Bayne, P. C. Fontaine, and C. J. Ralph. 2014. “Reviving Common Standards in Point-Count Surveys for Broad Inference Across Studies.” <em>Condor</em> 116: 599–608. <a href="https://doi.org/10.1650/CONDOR-14-108.1">https://doi.org/10.1650/CONDOR-14-108.1</a>.</p>
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="roadsides.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook/js/app.min.js"></script>
<script src="libs/gitbook/js/lunr.js"></script>
<script src="libs/gitbook/js/plugin-search.js"></script>
<script src="libs/gitbook/js/plugin-sharing.js"></script>
<script src="libs/gitbook/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook/js/plugin-bookdown.js"></script>
<script src="libs/gitbook/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/psolymos/qpad-book/edit/master/09-extras.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["qpad-book.pdf", "qpad-book.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
